yt_dlp/extractor/goplay.py

```diff
@@ -77,16 +93,38 @@ def _real_extract(self, url):
 
         api = self._download_json(
             f'https://api.goplay.be/web/v1/videos/long-form/{video_id}',
-            video_id, headers={'Authorization': 'Bearer %s' % self._id_token})
+            video_id, headers={
+                'Authorization': 'Bearer %s' % self._id_token,
+                **self.geo_verification_headers(),
+            })
+
+        if 'manifestUrls' in api:
+            formats, subs = self._extract_m3u8_formats_and_subtitles(
+                api['manifestUrls']['hls'], video_id, ext='mp4', m3u8_id='HLS')
 
-        formats, subs = self._extract_m3u8_formats_and_subtitles(
-            api['manifestUrls']['hls'], video_id, ext='mp4', m3u8_id='HLS')
+        else:
+            assert 'ssai' in api, 'expecting Google SSAI stream'
+
+            ssai_content_source_id = api['ssai']['contentSourceID']
+            ssai_video_id = api['ssai']['videoID']
+
+            dai = self._download_json(
+                f'https://dai.google.com/ondemand/dash/content/{ssai_content_source_id}/vid/{ssai_video_id}/streams',
+                video_id, data=b'{"api-key":"null"}',
+                headers={'content-type': 'application/json'})
+
+            periods = self._extract_mpd_periods(dai['stream_manifest'], video_id)
+
+            # skip pre-roll and mid-roll ads
+            periods = [period for period in periods
+                       if '-ad-' not in period['id']]
+
+            formats, subtitles = self._merge_mpd_periods(periods)
 
         info_dict.update({
             'id': video_id,
             'formats': formats,
```

Comment:
```suggestion
            'formats': formats,
            'subtitles': subtitles,
```
Why are you discarding subs?

Reply:
Good question. This part comes from the current version of the extractor. I don't think GoPlay currently offers any subtitles (except for some subtitles embedded in the video). The streams I've tried don't have them, so it doesn't make a practical difference.

I've now added the subtitles to the infodict, in case they start offering subtitles in the future.